sound/soc/sof/intel/hda-pcm.c

```diff
@@ -88,7 +88,7 @@ int hda_dsp_pcm_hw_params(struct snd_soc_component *scomp,
 			  struct snd_pcm_hw_params *params,
 			  struct sof_ipc_stream_params *ipc_params)
 {
-	struct snd_sof_dev *sdev = dev_get_drvdata(scomp->dev->parent);
+	struct snd_sof_dev *sdev = snd_sof_get_sof_dev(scomp->dev);
```

Comment:
@ranj063 Have you checked whether we could go even step further and not use "sdev" at all in these platform-specific functions? I see you've cleaned up sdev usage in many places already, so I'm assuming that these are still needed here.

Reply:
@kv2019i yes I thought about this. The main reason I didnt do this is because of the register reads/writes that the platform-specific code does. I suppose I could add abstracted methods for these as well but do you think its necessary? This is low-level code anyway no?

Reply:
@ranj063 no, no, just asking whether you've looked into this. And yeah, I now looked better at hda-pcm.c as an example, and yeah, I agreee adding abstractions here does not make sense. The low-level functions have to understand the device as a whole entity (not split into MFD functions, the hw is a combined device after all). So just asking.